common/workunit/workunit.cpp

```diff
@@ -2657,85 +2660,192 @@ cost_type aggregateCost(const IConstWorkUnit * wu, const char *scope, bool exclu
         }
         return totalCost;
     }
+};
+
+GlobalStatisticCollection::GlobalStatisticCollection() : aggregateKindsMapping(aggregateKindStatistics)
```

Comment:
I was thinking that we'd have a single location for all the stats being gathered and aggregated.      I was trying to avoid having one class to gather and then have another for aggregation - and then having to copy stats between the two.

Reply:
I think one of the other reasons it was more generalized (not just a aggregator), was that there was a separate future intent for it to manage the gathered stat. collections - and publish them en masse periodically (rather than end of sg as it is now).
That is a separate problem - updating Dali stats too frequently when there are many small quick running subgraphs.

Reply:
ok.  That could be achieved more simply by saving away a list of blobs to commit, and writing them all in one transaction.

I'm not sure if this is worth changing now - I will think about it, but it does complicate the implementation by combining two different concepts and needing to worry about pruning results etc.  It makes it harder to understand the responsibilities and implementation of the class etc.